sympy/combinatorics/pc_groups.py

```diff
@@ -41,27 +48,112 @@ def is_prime_order(self):
                 return False
         return True
 
+    def is_consistent(self):
+        if self.power_exponents.sort() == self.relative_orders.sort():
+            return True
+        return False
+
+    def _reorder_word(self, w):
+        '''
+        To find the minimal uncollected subword,
+        the word has to be reordered in the same order
+        as that of the polycyclic generating sequence.
+        '''
+        if not isinstance(w, FreeGroupElement):
+            raise ValueError("The word must be an instance of FreeGroupElement")
+        k = w.letter_form_elm
+        k.sort()
+        reordered_word = self.fp_group.identity
+        for x in k:
+            reordered_word *= x
+        return reordered_word
+
+    def minimal_uncollected_subwords(self, w):
+        # This store the subword along with the
+        # type and the exponent of the subword.
+        subword_dict = {}
+        arr_form = list(w.array_form)
+        word_index = 0
+
+        for i in range(0, len(arr_form)):
+            if i != (len(arr_form) - 1):
+                # Stores the first, last index and the
+                # type of the subword
+                index_arr = []
+                current_index = word_index
+                if arr_form[i+1][1] > 0:
+                    subword = arr_form[i][0]**arr_form[i][1]*arr_form[i+1]
+                    # subword of the type `x**k*y`.
+                    index_arr.append(1)
+                else:
+                    subword = arr_form[i][0]**arr_form[i][1]*arr_form[i+1]**-1
+                    # subword of the type `x**k*y**-1`.
+                    index_arr.append(2)
+                # Append the exponent `k`.
+                index_arr.append(arr_form[i][1])
+                word_index += abs(arr_form[i][1])
+                subword_dict[subword] = index_arr
+
+            # Check for subwords of the type `x**k`.
+            if not arr_form[i][1] in self.power_exponents:
+                index_arr = []
+                subword = arr_form[i][0]**arr_form[i][1]
+                # Subword of the type `x**k`.
+                index_arr.append(0)
+                current_index += abs(arr_form[i][1])
+                subword_dict[subword] = index_arr
+
+        return subword_dict
+
+    def _find_index_range(self, elem, w):
```

Comment:
This looks like it belongs in the `free_groups.py` among other `FreeGroupElement` methods - have you checked that the method you want doesn't already exist?

Reply:
So, have you checked there isn't a method that does this?